nipoppy_cli/nipoppy/data/examples/sample_global_config-all_pipelines.json

```diff
@@ -0,0 +1,161 @@
+{
+    "DATASET_NAME": "<DATASET_NAME>",
+    "VISITS": [
+        "<VISIT_LABEL>"
+    ],
+    "SESSIONS": [
+        "ses-<SESSION_LABEL>"
+    ],
+    "CONTAINER_CONFIG": {
+        "ARGS": [
+            "--cleanenv"
+        ]
+    },
+    "GLOBALS": {
+        "[[HEUDICONV_HEURISTIC_FILE]]": "<PATH_TO_HEURISTIC_FILE>",
+        "[[DCM2BIDS_CONFIG_FILE]]": "<PATH_TO_CONFIG_FILE>",
+        "[[FREESURFER_LICENSE_FILE]]": "<PATH_TO_FREESURFER_LICENSE_FILE>",
+        "[[TEMPLATEFLOW_HOME]]": "<PATH_TO_TEMPLATEFLOW_DIRECTORY>"
+    },
+    "BIDS_PIPELINES": [
+        {
+            "NAME": "heudiconv",
+            "VERSION": "0.12.2",
+            "STEP": "prepare",
+            "CONTAINER": "heudiconv_0.12.2.sif",
+            "URI": "docker://nipy/heudiconv:0.12.2",
+            "INVOCATION": {
+                "overwrite": true
+            }
+        },
+        {
+            "NAME": "heudiconv",
+            "VERSION": "0.12.2",
+            "STEP": "convert",
+            "CONTAINER": "heudiconv_0.12.2.sif",
+            "URI": "docker://nipy/heudiconv:0.12.2",
+            "INVOCATION": {
+                "heuristic": "[[HEUDICONV_HEURISTIC_FILE]]",
+                "overwrite": true,
+                "minmeta": true
+            },
+            "CONTAINER_CONFIG": {
+                "ARGS": [
+                    "--bind",
+                    "[[HEUDICONV_HEURISTIC_FILE]]"
+                ]
+            }
+        },
```

Comment:
It feels odd to have multiple steps listed as separate pipelines... I understand why you are doing it, but this can't happen with bigger processing pipelines. This also has separate `CONTAINER` AND `CONTAINER_CONFIG` which looks confusing. 
Why not make `STEPS` as nested block that can share `NAME`, `VERSION` etc. but have independent `INVOCATION`